pkg/lightning/restore/check_info.go

```diff
@@ -372,9 +365,19 @@ func (rc *Controller) SchemaIsValid(ctx context.Context, tableInfo *md.MDTableMe
 			}
 		default:
 			msgs = append(msgs, fmt.Sprintf("file '%s' with unknown source type '%s'", dataFileMeta.Path, dataFileMeta.Type.String()))
+			return msgs, nil
 		}
-		err = parser.ReadRow()
-		if err != nil {
+		defer parser.Close()
+
+		if err = parser.ReadRow(); err != nil {
+			if errors.Cause(err) == io.EOF {
+				msg := fmt.Sprintf("file '%s' doesn't have a data row", dataFileMeta.Path)
+				if dataFileMeta.Type == md.SourceTypeCSV {
+					msg += fmt.Sprintf(" (first row is parsed as header: %v)", rc.cfg.Mydumper.CSV.Header)
```

Comment:
I think we should emit this message if `rc.cfg.Mydumper.CSV.Header = false`. BTW, if the source file is empty or only contains the header line, I think it's better to skip this check instead of returning an error. What do you think @3pointer